src/box/lua/net_box.lua

```diff
@@ -243,12 +245,17 @@ local function create_transport(host, port, user, password, callback,
     ::stop::
             send_buf:recycle()
             recv_buf:recycle()
+            on_send_buf_empty:broadcast()
             worker_fiber = nil
         end)
     end
 
     local function stop()
         if not is_final_state[state] then
+            is_closing = true
+            while (send_buf:size() ~= 0) do
+                on_send_buf_empty:wait()
+            end
             set_state('closed', E_NO_CONNECTION, 'Connection closed')
```

Comment:
If two fibers enter here, they will both update the state once they're done waiting. Please fix.

Reply:
fixed